scipy/datasets/tests/test_data.py

```diff
@@ -71,3 +63,32 @@ def test_electrocardiogram(self):
         # hash check
         assert _has_hash(os.path.join(data_dir, "ecg.dat"),
                          registry["ecg.dat"])
+
+
+def test_clear_cache():
+    # Use Dummy path
+    dummy_basepath = "dummy_cache_dir"
+    os.makedirs(dummy_basepath, exist_ok=True)
```

Comment:
Can we use a `tmp_path` fixture from `pytest` to avoid polluting the local working space when this test fails?

To be fair, the pollution ends up in `./build-install/lib/python3/dist-packages/dummy_cache_dir`, which I don't usually care about much, but still seems best practice to just direct temporary test stuff to a conventional temporary space on the machine.

This is also relates somewhat to my suggestion above re: adding arguments to the `_clear_cache` function only for testing--couldn't we just use `scipy-data` itself once we've changed the working directory context to whatever temporary path `pytest` uses with the fixture?

Reply:
The current `tmp_path` version LGTM, resolving.